src/go/rpk/pkg/cli/cmd/cluster/config/lint.go

```diff
@@ -21,6 +21,26 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
+// clusterConfigN represents a redpanda configuration.
+type clusterConfigN map[string]yaml.Node
+
+// A custom unmarshal is needed because go-yaml parse "YYYY-MM-DD" as a full
+// timestamp, writing YYYY-MM-DD HH:MM:SS +0000 UTC when encoding, so we are
+// going to treat timestamps as strings.
+// See: https://github.com/go-yaml/yaml/issues/770
```

Comment:
Probably not the most necessary to duplicate this comment, but it's also not necessary to remove at this point